### DISCUSSION TITLE
[Help]: RTL list image lazy loading not working

### DISCUSSION THREAD

**Discussion by tariqinaam** [davidjerleke/embla-carousel#736] [Help] [answered]
### Summary

Hi,
I have nextjs project where I am using embla with RTL support. images on the far left are loading and the one in view doesnt load. When scrolling to the end of list and come back, it loads images.
I have added dir="rtl" in section root div as well. 
Is there anything I am missing here.?

`export default function CollectionSlider({ row, onUserPlayRequest, setSpotlightData, nowShowing = null, selectedLanguage }) {

    const [emblaRef, emblaApi] = useEmblaCarousel({ dragFree: true, duration: 35, direction: "rtl", slidesToScroll: "auto" }, [
        WheelGesturesPlugin(),
    ])
    return row?.pinned?.length > 0 || row?.programmes?.length > 0 ? (
        <div
            className="z-10 text-white relative first:mt-10 md:mt-10 mt-6"
        >

            <div className="pb-4 font-medium text-gray-200 sm:text-[21px] lg:text-[17px]">{`${row.label}`}</div>
            <div className="embla overflow-hidden" ref={emblaRef}>
                <div className="embla__container flex">
                    {(nowShowing) && <LiveSliderItem key={0} nowShowing={nowShowing} />}

                    {row.pinned?.map((item, i) => {
                        return <CollectionSliderItem item={item} row={row} key={i} onUserPlayRequest={onUserPlayRequest} setSpotlightData={setSpotlightData} selectedLanguage={selectedLanguage} />;
                    })}
                    {row.programmes?.map((item, i) => {
                        return <CollectionSliderItem item={item} row={row} key={i} onUserPlayRequest={onUserPlayRequest} setSpotlightData={setSpotlightData} selectedLanguage={selectedLanguage} />;
                    })}
                </div>
            </div>
        </div>
    ) : (
        <></>
    )}`

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment]
@tariqinaam thanks for your question. What version of Embla are you using? Additionally, please create a minimal CodeSandbox that reproduces the problem. I can't debug fragments of code. Thanks!

**Reply by tariqinaam** [reply]
Sorry, I should have added in initial request.
I am using 
`"embla-carousel-react": "^8.0.0-rc11",
    "embla-carousel-wheel-gestures": "^8.0.0-rc04",`

**Reply by davidjerleke** [reply]
@tariqinaam thanks, as mentioned in my [previous comment](https://github.com/davidjerleke/embla-carousel/discussions/736#discussioncomment-8367859):

> Please create a minimal CodeSandbox that reproduces the problem. I can't debug fragments of code. Thanks!

**Reply by tariqinaam** [reply]
here is sandbox for it to replicate
https://codesandbox.io/p/devbox/embla-rtl-images-x6qvzr
you need to open preivew in new window to see the issue. thanks

**Reply by sarussss** [reply]
Hi @tariqinaam, if I'm not mistaken, you use img tag with loading="lazy" attribute which is supported by browser so loading this image is not related to Embla library.

**Reply by davidjerleke** [reply]
@tariqinaam the only thing Embla does to the slide container is that it applies `translate3D()` on the **carousel container** to create the scroll effect. That's all. You're using an `AkamaiImageLoader` loader on your image component. I have no idea what that does under the hood but please make sure that's not the culprit. The bug could be on their side. Maybe they don't support `RTL`, or maybe it needs to be configured? 

Best,
David

**Comment by sirajtahra** [comment]
facing the same issue +1

**Reply by davidjerleke** [reply]
@sirajtahra regardless, it's not likely to be Embla related. Please look for the problem in other tools. I've tested lazy loading images with RTL by setting [this example](https://www.embla-carousel.com/examples/predefined/#lazy-load) to RTL and it's working as expected.

**Reply by sirajtahra** [reply]
I agree, it might not be embla related, the issue seems in the Nextjs Image component which for some reason is failing to detect that the image is actually within the viewport

**Reply by sirajtahra** [reply]
@davidjerleke any thoughts on this?

**Reply by davidjerleke** [reply]
@sirajtahra I would try removing libraries/tools/components until it starts working as expected to find what's causing this. As I mentioned I've tested lazy loading images with `RTL` by setting [this example](https://www.embla-carousel.com/examples/predefined/#lazy-load) to `RTL` and it's working as expected. 

So something else is probably causing this. If you can find which component it is, for example if it's the `NextJS image component` or `AkamaiImageLoader`, I would create a bug report so the NextJS or team who's responsible for that product/component can investigate it.

Best,
David

**Reply by sirajtahra** [reply, marked as answer]
@davidjerleke After investigation, the issue has to do with the intersection observer API not working as expected when `<html dir="rtl">` in the latest chrome version, the issue does not show if you are using other browsers (Safari, Firefox) or when the direction is set to `ltr`.

Using a previous chrome version fixed the issue. It has nothing to do with Nextjs either, rather with the native `lazy` property for the `img` component, which relies on the intersection observer. source: https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading

Thank you @davidjerleke for your quick response and understanding.

Chome Version with issue: 121.0 and above
Chrome version without issue: 120.0 or below

Chromium bug link: https://issues.chromium.org/issues/41495751?pli=1

**Reply by davidjerleke** [reply]
@sirajtahra nice work finding the culprit 👍! I haven't updated to the latest Chrome version so that explains why I couldn't reproduce it. If you don't mind, you can add your Chrome version [to you comment](https://github.com/davidjerleke/embla-carousel/discussions/736#discussioncomment-8476971) so devs experiencing the same problem can find it.

**Reply by kc980602** [reply]
@sirajtahra Did you manage to find a workaroud, seems chromium bug fix will not be release to public until mid-April with Chrome version 124 .
https://issues.chromium.org/issues/41495751#comment29

**Reply by sirajtahra** [reply]
@kc980602 Unfortunately I couldn't find a work around, as it seems to be a pure chromium engine bug which has to do with the core.
